pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/TimeSeriesRequestHandler.java

```diff
@@ -0,0 +1,172 @@
+      RequestContext requestContext) {
+    RangeTimeSeriesRequest timeSeriesRequest = null;
+    try {
+      timeSeriesRequest = buildRangeTimeSeriesRequest(lang, rawQueryParamString);
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+    TimeSeriesLogicalPlanResult logicalPlanResult = _queryEnvironment.buildLogicalPlan(timeSeriesRequest);
+    TimeSeriesDispatchablePlan dispatchablePlan = _queryEnvironment.buildPhysicalPlan(timeSeriesRequest, requestContext,
+        logicalPlanResult);
+    return _queryDispatcher.submitAndGet(requestContext, dispatchablePlan, timeSeriesRequest.getTimeout().toMillis(),
+        new HashMap<>());
+  }
+
+  @Override
+  public Map<Long, String> getRunningQueries() {
+    // TODO: Implement this.
+    return Map.of();
+  }
+
+  @Override
+  public boolean cancelQuery(long queryId, int timeoutMs, Executor executor, HttpClientConnectionManager connMgr,
+      Map<String, Integer> serverResponses)
+      throws Exception {
+    // TODO: Implement this.
+    return false;
+  }
+
+  private RangeTimeSeriesRequest buildRangeTimeSeriesRequest(String language, String queryParamString)
+      throws URISyntaxException {
+    List<NameValuePair> pairs = URLEncodedUtils.parse(
+        new URI("http://localhost?" + queryParamString), "UTF-8");
+    String query = null;
+    Long startTs = null;
+    Long endTs = null;
+    String step = null;
+    String timeoutStr = null;
+    for (NameValuePair nameValuePair : pairs) {
+      switch (nameValuePair.getName()) {
+        case "query":
+          query = nameValuePair.getValue();
+          break;
+        case "start":
+          startTs = Long.parseLong(nameValuePair.getValue());
+          break;
+        case "end":
+          endTs = Long.parseLong(nameValuePair.getValue());
+          break;
+        case "step":
+          step = nameValuePair.getValue();
+          break;
+        case "timeout":
+          timeoutStr = nameValuePair.getValue();
+          break;
+        default:
+          /* Okay to ignore unknown parameters since the language implementor may be using them. */
+          break;
+      }
+    }
+    Preconditions.checkNotNull(query, "Query cannot be null");
+    Preconditions.checkNotNull(startTs, "Start time cannot be null");
+    Preconditions.checkNotNull(endTs, "End time cannot be null");
+    Duration timeout = Duration.ofMillis(_brokerTimeoutMs);
+    if (StringUtils.isNotBlank(timeoutStr)) {
+      timeout = HumanReadableDuration.from(timeoutStr);
+    }
+    // TODO: Pass full raw query param string to the request
+    return new RangeTimeSeriesRequest(language, query, startTs, endTs, getStepSeconds(step), timeout, queryParamString);
+  }
+
+  public static Long getStepSeconds(@Nullable String step) {
+    if (step == null) {
+      return DEFAULT_STEP_SECONDS;
+    }
+    try {
+      return Long.parseLong(step);
+    } catch (NumberFormatException ignored) {
```

Comment:
Why are we ignoring the exception? Will we revert to default in case of invalid step time?

Reply:
Oh because the duration passed by the client could be something like `step=10s`, `step=10`. In the former case it's very easy to know what the expectation. In the latter case, we right now assume that the default unit the user is targeting is seconds.